### ISSUE TITLE
Upgrading from 4.0.2 to 4.0.3 seems to break link calls inside subflows

### ISSUE THREAD

**Issue by natcl** [node-red/node-red#4891]
### Current Behavior

We updated one of our projects from 4.0.2 to 4.0.3 and noticed that some link calls within subflows had lost the association and were showing "None" with the editor issuing a warning upon deploy.

Our flow is quite complex so difficult to post here but let me know if I can add something to help.

### Example flow

```
paste your flow here
```

### Environment

- Node-RED version: 4.0.3
- Node.js version: 20.17.0
- npm version:
- Platform/OS:
- Browser:

**Comment by natcl** [comment]
@knolleary thinks it's probably related to [#4857 Fix subflow outbound-link filter](https://github.com/node-red/node-red/pull/4857)

**Comment by knolleary** [comment]
To clarify - you have link call nodes inside subflows. Where are the link in nodes they are calling? In the same subflow? Or on a regular tab? Somewhere else?

**Comment by natcl** [comment]
The links that are called are in the parent flow (top level)

**Comment by GogoVega** [comment]
In my fixes mentioned above, links inside a subflow can only interact with other links into the same subflow.

**Comment by knolleary** [comment]
That is true if link in/out nodes, but the link call node should be able to call outside of the subflow

**Comment by Steve-Mcl** [comment]
Form memory, we allowed link-call to call out to its parent and if a matching link was found it was used.  This was a known.  

@GogoVega are you saying your PR changed this behaviour?

UPDATE:
Original design discussion here: https://github.com/node-red/node-red/issues/3441#issuecomment-1052265984

**Comment by knolleary** [comment]
Yes, I overlooked that scenario when I reviewed/updated the pr. My bad. Will fix tomorrow

**Comment by Steve-Mcl** [comment]
I was convinced I wrote tests for this. And I did. But it seems I disabled the one subflow test due to an issue with loading flows with subflows. bah humbug  :( 

https://github.com/node-red/node-red/pull/3463/files#diff-f138a40cc1073b0d7bebd7c4e781ee24d66dccea378578194808abaec76a7f17R185

**Comment by GogoVega** [comment]
> @GogoVega are you saying your PR changed this behaviour?

No, the regex matched all 3 links and I didn't change it.

**Comment by knolleary** [comment]
@Steve-Mcl this is an editor thing, not runtime, so your tests wouldn't have been touched by this change.

**Comment by Steve-Mcl** [comment]
Ah, ok. that makes more sense. Phew. I thought for a moment users flows were affected in the runtime.

**Comment by natcl** [comment]
Thanks :)

**Comment by natcl** [comment]
Hello ! I just upgraded to 4.0.4 and still get the popup
Here's a screenshot for reference:
<img width="1206" alt="image" src="https://github.com/user-attachments/assets/9c57c61b-4fac-49aa-b109-8152af820c45">

Let me know if I can help with more details !

**Comment by natcl** [comment]
Here's a minimal flow to reproduce:
```
[
    {
        "id": "5b796fcc5c17551c",
        "type": "tab",
        "label": "Main",
        "disabled": false,
        "info": "",
        "env": []
    },
    {
        "id": "c40de3cc6c5254e9",
        "type": "subflow",
        "name": "Subflow 1",
        "info": "",
        "in": [
            {
                "x": 40,
                "y": 40,
                "wires": [
                    {
                        "id": "cee9f1f2d5782971"
                    }
                ]
            }
        ],
        "out": [
            {
                "x": 380,
                "y": 40,
                "wires": [
                    {
                        "id": "cee9f1f2d5782971",
                        "port": 0
                    }
                ]
            }
        ]
    },
    {
        "id": "cee9f1f2d5782971",
        "type": "link call",
        "z": "c40de3cc6c5254e9",
        "name": "",
        "links": [
            "87e66d8907eec845"
        ],
        "linkType": "static",
        "timeout": "30",
        "x": 190,
        "y": 40,
        "wires": [
            []
        ]
    },
    {
        "id": "87e66d8907eec845",
        "type": "link in",
        "z": "5b796fcc5c17551c",
        "name": "Foo",
        "links": [],
        "x": 125,
        "y": 100,
        "wires": [
            [
                "90d01a1d8137758b",
                "6339d6f90ab101cf"
            ]
        ]
    },
    {
        "id": "90d01a1d8137758b",
        "type": "debug",
        "z": "5b796fcc5c17551c",
        "name": "debug 1",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "false",
        "statusVal": "",
        "statusType": "auto",
        "x": 380,
        "y": 100,
        "wires": []
    },
    {
        "id": "6339d6f90ab101cf",
        "type": "link out",
        "z": "5b796fcc5c17551c",
        "name": "Return",
        "mode": "return",
        "links": [],
        "x": 285,
        "y": 140,
        "wires": []
    },
    {
        "id": "b60293ac83a46c74",
        "type": "subflow:c40de3cc6c5254e9",
        "z": "5b796fcc5c17551c",
        "name": "",
        "x": 280,
        "y": 240,
        "wires": [
            []
        ]
    }
]
```

**Comment by knolleary** [comment]
Will be fixed in 4.0.5

**Comment by natcl** [comment]
Thanks, 404 is not a lucky number 😛
